androidHyperskillApp/src/main/res/layout/item_project.xml

```diff
@@ -0,0 +1,139 @@
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:textAppearance="@style/TextAppearance.AppTheme.Caption"
+            android:textColor="@color/color_on_surface_alpha_60"
+            android:drawablePadding="6dp"
+            app:drawableStartCompat="@drawable/ic_star"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintTop_toTopOf="parent"
+            tools:text="4.7"/>
+
+        <TextView
+            android:id="@+id/projectTitle"
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            android:textAppearance="@style/TextAppearance.AppTheme.Body1"
+            android:maxLines="2"
+            android:ellipsize="end"
+            android:layout_marginTop="12dp"
+            app:layout_constraintTop_toBottomOf="@+id/projectLevel"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintEnd_toEndOf="parent"
+            tools:text="Simple Chatty Bot (Python)Simple Chatty Bot (Python)Simple Chatty Bot (Python)Simple Chatty Bot (Python)"/>
+
+        <TextView
+            android:id="@+id/projectDuration"
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            android:textAppearance="@style/TextAppearance.AppTheme.Caption"
+            android:textColor="@color/color_on_surface_alpha_60"
+            android:layout_marginTop="4dp"
+            android:layout_marginBottom="12dp"
+            app:layout_constraintTop_toBottomOf="@+id/projectTitle"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintEnd_toEndOf="parent"
+            tools:text="136 hours"
+            />
+
+        <androidx.constraintlayout.helper.widget.Flow
+            android:id="@+id/projectTags"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            app:constraint_referenced_ids="projectCompletedTextView,projectSelectedTextView,projectIdeRequiredTextView,projectBestRatingTextView,projectFastestToCompleteTextView"
+            app:flow_horizontalAlign="start"
+            app:flow_horizontalBias="0"
+            app:flow_verticalGap="4dp"
+            app:flow_wrapMode="chain"
+            app:flow_horizontalGap="8dp"
+            app:flow_horizontalStyle="packed"
+            android:layout_marginTop="@dimen/project_selection_list_item_tags_top_margin"
+            app:layout_constraintTop_toBottomOf="@+id/projectDuration"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintEnd_toEndOf="parent"
+            />
+
+        <TextView
+            android:id="@+id/projectCompletedTextView"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            style="@style/Badge.Green"
+            android:text="@string/completed_label"
```

Comment:
We also have `Completed` and `IDE required` badges on the study plan screen. Is it relevant for Android development to create reusable layouts? For example, I mean smth like include here predefined `current`/`ideRequired` badge layout, now it seems like code duplication. Or create some base badge layout, that could be configurable.

Reply:
Badges are just text views with different colors and texts.
At the moment colors and paddings are handled by styles.
Only text parameters should be configured.
But you are right: it is more useful to create a set of predefined views, which can be used without text setting.